libs/langchain/langchain/utilities/pubmed.py

```diff
@@ -128,39 +145,31 @@ def retrieve_article(self, uid: str, webenv: str) -> dict:
                     raise e
 
         xml_text = result.read().decode("utf-8")
-
-        # Get title
-        title = ""
-        if "<ArticleTitle>" in xml_text and "</ArticleTitle>" in xml_text:
-            start_tag = "<ArticleTitle>"
-            end_tag = "</ArticleTitle>"
-            title = xml_text[
-                xml_text.index(start_tag) + len(start_tag) : xml_text.index(end_tag)
-            ]
-
-        # Get abstract
-        abstract = ""
-        if "<AbstractText>" in xml_text and "</AbstractText>" in xml_text:
-            start_tag = "<AbstractText>"
-            end_tag = "</AbstractText>"
-            abstract = xml_text[
-                xml_text.index(start_tag) + len(start_tag) : xml_text.index(end_tag)
-            ]
-
-        # Get publication date
-        pub_date = ""
-        if "<PubDate>" in xml_text and "</PubDate>" in xml_text:
-            start_tag = "<PubDate>"
-            end_tag = "</PubDate>"
-            pub_date = xml_text[
-                xml_text.index(start_tag) + len(start_tag) : xml_text.index(end_tag)
+        text_dict = self.parse(xml_text)
+        return self._parse_article(uid, text_dict)
+
+    def _parse_article(self, uid: str, text_dict: dict) -> dict:
+        ar = text_dict["PubmedArticleSet"]["PubmedArticle"]["MedlineCitation"][
+            "Article"
+        ]
+        summary = "\n".join(
+            [
+                f"{txt['@Label']}: {txt['#text']}"
+                for txt in ar.get("Abstract", {}).get("AbstractText", [])
+                if "#text" in txt and "@Label" in txt
             ]
+        )
+        a_d = ar.get("ArticleDate", {})
+        pub_date = "-".join(
+            [a_d.get("Year", ""), a_d.get("Month", ""), a_d.get("Day", "")]
+        )
 
-        # Return article as dictionary
-        article = {
+        return {
             "uid": uid,
-            "title": title,
-            "summary": abstract,
-            "pub_date": pub_date,
+            "Title": ar.get("ArticleTitle", ""),
+            "Published": pub_date,
```

Comment:
nit: why title case instead of snake_case, given that it was already in snake_case before? (it's common to use snake_case for keys while on the backend (at least in python) and convert to and from CamelCase when interacting w/ FE)

Reply:
Here I change the naming convention from snake_case to the TItle because from this place the keys are exposed to LangChain. In the LangChain we usually finish by exposing meta information/content as a formatted string. Something like "Title: ...\nAuthors:\nJohn...\nFred...\n\nSummary:\n..." Capital letters are important for LLM in this string.
I don't want any additional case transformation `inside` LangChain.

So, All `key naming` transformation is happening inside the integration "reader". No `key naming` transformations inside LangChain (str <-> dict <-> Document).

Note: I didn't change "uid" key. Let's do it now. Reasons: 1. it is the only field which value is passed without transformations. 2. It is `internal id` == `implementation detail`. I leave it only for a case when we need to backtrace a document in PubMed. In this case, we need the original name. 3. this case is not exposed in Document format but only in dict format (in load() not in run() nor PubMedDocumentLoader.load())